libraries/botbuilder-azure/botbuilder/azure/blob_storage.py

```diff
@@ -63,24 +62,31 @@ def __init__(self, settings: BlobStorageSettings):
         return items
 
     async def write(self, changes: Dict[str, object]):
+        if changes is None:
+            raise Exception("Changes are required when writing")
+        if not changes:
+            return
+
         self.client.create_container(self.settings.container_name)
         self.client.set_container_acl(
             self.settings.container_name, public_access=PublicAccess.Container
         )
 
-        for name, item in changes.items():
-            e_tag = (
-                None if not hasattr(item, "e_tag") or item.e_tag == "*" else item.e_tag
-            )
-            if e_tag:
-                item.e_tag = e_tag.replace('"', '\\"')
+        for (name, item) in changes.items():
+            e_tag = item.e_tag if hasattr(item, "e_tag") else item.get("e_tag", None)
```

Comment:
was there any reason to change previous form? This version has more conditionals to achieve the same thing

Reply:
Mostly so I could add `item.get("e_tag", None)` to account for non-`StoreItem`s. It was starting to look really sloppy and tough to read if I added that into the conditional. Open to other ideas, though.